src/main/java/org/zalando/putittorest/RestClientAutoConfiguration.java

```diff
@@ -27,7 +28,7 @@
 public class RestClientAutoConfiguration {
 
     @Bean
-    public RestClientPostProcessor restClientPostProcessor(final PluginResolver resolver) {
+    public static RestClientPostProcessor restClientPostProcessor(final ObjectFactory<PluginResolver> resolver) {
```

Comment:
The added `static` modifier is my poor attempt at getting rid of the following `WARN` logged on start:

```
ConfigurationClassPostProcessor  : Cannot enhance @Configuration bean definition 'restClientPostProcessor' since its singleton instance has been 
created too early. The typical cause is a non-static @Bean method with a BeanDefinitionRegistryPostProcessor return type: Consider declaring such methods as 'static'.
```
I had a quick look into this. It seems as `restClientPostProcessor` is registered as a factory method on the `RestClientAutoConfiguration`, the post processor also gets assigned the "I-am-a-configuration-flag", which is weird. As we know that it is not a configuration, one can probably ignore the warning.